lms/djangoapps/verify_student/signals.py

```diff
@@ -14,11 +14,11 @@ def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable
     Catches the signal that a course has been published in Studio and
     sets the verification deadline date to a default.
     """
-    try:
-        deadline = VerificationDeadline.objects.get(course_key=course_key)
-        if deadline and not deadline.deadline_is_explicit:
-            course = modulestore().get_course(course_key)
-            if course and deadline.deadline != course.end:
+    course = modulestore().get_course(course_key)
+    if course:
+        try:
+            deadline = VerificationDeadline.objects.get(course_key=course_key)
+            if not deadline.deadline_is_explicit and deadline.deadline != course.end:
                 VerificationDeadline.set_deadline(course_key, course.end)
-    except ObjectDoesNotExist:
-        pass
+        except ObjectDoesNotExist:
+            VerificationDeadline.set_deadline(course_key, course.end)
```

Comment:
The original logic would not create a VerificationDeadline record if one didn't already exist (and it only existed if the user manually set it through CAT).

This logic will create one if one doesn't exist, but we need to know what date to set.  So I moved when we grab the course to happen first and then the rest of the logic is nested within that.